src/librustc_metadata/rmeta/table.rs

```diff
@@ -93,24 +93,34 @@ impl<T: Encodable> FixedSizeEncoding for Option<Lazy<T>> {
     }
 }
 
-impl<T: Encodable> FixedSizeEncoding for Option<Lazy<[T]>> {
-    fixed_size_encoding_byte_len_and_defaults!(u32::BYTE_LEN * 2);
+macro_rules! meta_body {
```

Comment:
`impl_fixed_size_encoding_for_usize_meta` would be a better name, but also, have you tried this?
```diff
-impl<T: Encodable> FixedSizeEncoding for Option<Lazy<[T]>> {
+impl<T: ?Sized + Encodable + LazyMeta<Meta = usize>> FixedSizeEncoding for Option<Lazy<T>> {
```
Although perhaps that is still considered overlapping, I'm not sure of the coherence rules around this.